src/tests/dynamo0p3_test.py

```diff
@@ -2337,6 +2330,96 @@ def test_bc_kernel_field_only(monkeypatch):
                 in str(excinfo))
 
 
+def test_operator_bc_kernel():
+    ''' Tests that a kernel with a particular name is recognised as a
+    kernel that applies boundary conditions to operators and that
+    appropriate code is added to support this. '''
+    _, invoke_info = parse(os.path.join(BASE_PATH,
+                                        "12.4_enforce_op_bc_kernel.f90"),
+                           api="dynamo0.3")
+    psy = PSyFactory("dynamo0.3").create(invoke_info)
+    generated_code = str(psy.gen)
+    print generated_code
+    output1 = "INTEGER, pointer :: boundary_dofs(:,:) => null()"
+    assert output1 in generated_code
+    output2 = "boundary_dofs => op_a_proxy%fs_to%get_boundary_dofs()"
+    assert output2 in generated_code
+    output3 = (
+        "CALL enforce_operator_bc_code(cell, nlayers, op_a_proxy%ncell_3d, "
+        "op_a_proxy%local_stencil, ndf_any_space_1_op_a, "
+        "ndf_any_space_2_op_a, boundary_dofs)")
+    assert output3 in generated_code
+
+
+def test_operator_bc_kernel_fld_err(monkeypatch):
+    ''' test that we reject the recognised operator boundary conditions
+    kernel if its argument is not an operator '''
+    _, invoke_info = parse(os.path.join(BASE_PATH,
+                                        "12.4_enforce_op_bc_kernel.f90"),
+                           api="dynamo0.3")
+    for dist_mem in [False, True]:
+        psy = PSyFactory("dynamo0.3",
+                         distributed_memory=dist_mem).create(invoke_info)
+        schedule = psy.invokes.invoke_list[0].schedule
+        loop = schedule.children[0]
+        call = loop.children[0]
+        arg = call.arguments.args[0]
+        # Monkeypatch the argument object so that it thinks it is a
+        # field rather than an operator
+        monkeypatch.setattr(arg, "_type", value="gh_field")
+        with pytest.raises(GenerationError) as excinfo:
+            _ = psy.gen
+        assert ("Expected a LMA operator from which to look-up boundary dofs "
+                "but kernel enforce_operator_bc_code has no such argument") \
+            in str(excinfo)
+
+
+def test_operator_bc_kernel_multi_args_err():  # pylint: disable=invalid-name
```

Comment:
Shouldn't we also be testing that there is only one argument? This test implies it is valid for there to be more than one argument as long as only one of them is an operator.